src/Stratis.SmartContracts.CLR/SCL/ECRecover.cs

```diff
@@ -1,10 +1,34 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using NBitcoin;
+using NBitcoin.DataEncoders;
 using Stratis.SmartContracts;
+using Stratis.SCL.Base;
 
 namespace Stratis.SCL.Crypto
 {
     public static class ECRecover
     {
+        /// <summary>
+        /// Takes a message and signatures and recovers the list of addresses that produced the signatures.
+        /// </summary>
+        /// <param name="signatures">The list of signatures of the message.</param>
+        /// <param name="message">The message that was signed.</param>
+        /// <param name="addresses">If not <c>null</c> the addresses returned are intersected with these addresses.</param>
+        /// <returns>The list of addresses that produced the signatures and constrained to the list provided in <paramref name="addresses"/> if not <c>null</c>.</returns>
+        public static Address[] VerifySignatures(byte[] signatures, byte[] message, Address[] addresses = null)
```

Comment:
What do you think of the following:
- Split this into 2 methods instead of allowing a null param for addresses
- Using the Try pattern instead of returning null ie. `bool VerifySignatures(..., out Address[] verifiedAddresses)`

Reply:
We should also wrap everything in a try/catch (gross, I know, but NBitcoin especially likes to throw a lot) to prevent exceptions being propagated into the contract.

Reply:
Done.